src/ElfUtils/ElfFile.cpp

```diff
@@ -182,12 +189,38 @@ void ElfFileImpl<ElfT>::InitSections() {
   }
 }
 
+bool AddSymbol(const llvm::object::ELFSymbolRef& symbol_ref, ModuleSymbols& module_symbols,
+               std::string_view file_path) {
```

Comment:
maybe return std::optional here or ErrorMessageOr<SymbolInfo>. The caller can then move it by doing
*module_symbols.add_symbol_infos() = std::move(symbol_or_error.value());

Reply:
I cleaned up the method a bit and it and kept the adding of the symbol inside this method. Turns out the returned bool is really not needed.
Please take a look, if you still think the add_symbol_infos should go into the caller please let me know.

Reply:
personal preference: I would still rather have it return SymbolInfo, and do the add to container part on the caller side. And maybe rename it to CreateSymbolInfo(const llvm::object::ELFSymbolRef& symbol_ref). file_path seems to only be used for logging, if this function is moved to ElfFileImpl<ElfT> it can just access it from a field.

Reply:
added CreateSymbolInfo to ElfFileImpl